File: app/src/main/java/com/pgssoft/mvvm/viewmodels/RateViewModel.java
package com.pgssoft.mvvm.viewmodels;

import android.databinding.ObservableField;

import com.pgssoft.mvvm.model.database.Rate;
import com.pgssoft.mvvm.views.adapters.interfaces.handlers.RateItemHandler;

import java.util.Random;

/**
 * Created by bstokrocki on 30.01.2017.
 */
public class RateViewModel extends BaseListItemViewModel<Rate>{
    private RateItemHandler itemHandler;

    private ObservableField<Rate> rate;
    private final ObservableField<Boolean> isGoingUp;

    public RateViewModel(RateItemHandler itemHandler) {
        this.itemHandler = itemHandler;

        rate = new ObservableField<>();
        isGoingUp = new ObservableField<>(new Random().nextBoolean());
    }

    @Override
    public void setData(Rate rate) {
        this.rate.set(rate);
    }

    public ObservableField<Rate> getRate() {
        return rate;
    }

    public void itemClicked() {
        itemHandler.showRateDetails(rate.get());
    }

    public ObservableField<Boolean> getIsGoingUp() {
        return isGoingUp;
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/viewmodel/RateViewModel.kt
package com.pgssoft.kotlinplayground.viewmodel

import android.databinding.ObservableBoolean
import android.databinding.ObservableField
import com.pgssoft.kotlinplayground.model.db.Rate
import com.pgssoft.kotlinplayground.view.adapter.handlers.RateItemHandler
import java.util.*

/**
 * Created by bstokrocki on 30.01.2017.
 */
class RateViewModel(private val itemHandler: RateItemHandler) : BaseListItemViewModel<Rate>() {
    val rate = ObservableField<Rate>()
    val isGoingUp = ObservableBoolean(Random().nextBoolean())

    override fun setData(rate: Rate) {
        this.rate.set(rate)
    }

    fun itemClicked() {
        itemHandler.showRateDetails(rate.get())
    }
}
